---
title: "Analysis of MHV RNA to Detect EndoU Cleavage Sites"
author: "Rachel Ancar"
date: "11/03/2018"
output:
  html_document:
    code_folding: hide
    toc: yes
    toc_float: yes
  pdf_document:
    toc: yes
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_chunk$set(message = FALSE)
```


```{r, message = F, warning = F}

library(tidyverse)
library(readr)
library(dplyr)
library(ggplot2)
library(stringr)
library(cowplot)
library(RColorBrewer)
library(purrr)
library(shiny)
library(DT)
library(gghighlight)
library(rlang)
library(glue)

```

2´,3´-cyclic phosphate RNA-sequencing libraries were prepared from mouse macrophages: B6 (WT), IFNAR KO, RNaseL KO. These cells were either mock infected or infected with WT MHV virus (from Volker or Susan), MHV mutated to inactivate ns2 (phosphodiesterase) activity, or MHV mutated to inactivate nsp15 (endoribonuclease) activity for 9 and 12 hours. 

```{r, message = F, warning = F}

sample_info_table <- read_tsv("sample_info.txt", col_names = c("cell type", "virus type", "time"))
datatable(sample_info_table)

```

These libraries were processed to generate bedgraph files containing sites of 3´-cleavage and the associated dinucleotide in the MHV (+) strand RNA (mRNA) and (-) strand RNA (genomic). 

#Coverage plots for all cell types by type of viral infection

These plots show the normalized counts of captured 2´,3´-cyclic phosphates captured (% of total cDNA reads for each library) per nucleotide position in the the MHV (+) strand reads. The MHV (-) strand reads are displayed using un-normalized counts because there were very few detected cleavage events and those occured at very low frequency. 

```{r, message = F, warning = F}

#MHV (+) strand coverage plots 

data_dir_neg = "~/Dropbox (Hesselberth Lab)/Rachel_data/EndoU_project/viral_bg/neg"
data_files_neg = list.files(data_dir_neg, full.names = T)

read_file <- function(x) {
  df <- readr::read_tsv(x, col_names = c("chrom", "start", "end", "count", "normalized_count", "dinuc"))
  df$name <- basename(x)
  df
}

neg_table <- purrr::map_df(data_files_neg, read_file) %>%
  mutate(name = str_replace(name, ".mhv.neg.dinuc.bg", "")) %>%
  separate(name, into = c('cell', 'virus', 'time'), sep = '_')
datatable(neg_table)

#neg_plot <- ggplot(neg_table, aes(x = start, y = normalized_count)) +
    #geom_line(aes(color = time)) +
    #scale_colour_brewer(palette="Set1") + 
    #theme_cowplot() + 
    #facet_grid(virus ~ cell) +
    #theme(axis.text.x = element_text(angle = 60, hjust = 1)) +
    #labs(x="Position", y="Normalized counts") +
    #scale_x_continuous(breaks=seq(0, 30000, 5000)) + 
    #ggtitle("MHV (+) strand coverage plots")
#neg_plot

neg_plot <- ggplot(neg_table, aes(x = start, y = normalized_count, fill = time, width = 100)) +
    geom_bar(stat = "identity", position = 'identity') +
    scale_fill_brewer(palette="Set1") + 
    theme_cowplot() + 
    facet_grid(virus ~ cell) +
    theme(axis.text.x = element_text(angle = 60, hjust = 1)) +
    labs(x="Position", y="Normalized counts") +
    scale_x_continuous(breaks=seq(0, 30000, 5000)) + 
    ggtitle("MHV (+) strand coverage plots")
neg_plot

neg_plot <- neg_table %>%
  filter(cell == "B6") %>%
  filter(time != "012" & time != "09") %>%
  filter(virus == "MHVS") %>% 
  mutate(time = fct_relevel(time, "9", "12")) %>%
  #mutate(virus = fct_relevel(virus, "mock", "MHVV", "nsp15")) %>%
  ggplot(aes(x = start, y = normalized_count, fill = time, width = 25)) +
    geom_bar(stat = "identity", position = 'identity') +
    scale_fill_grey(start=0.8, end=0.2) +
    theme_cowplot() + 
    facet_grid(time ~ cell) +
    theme(axis.text.x = element_text(angle = 60, hjust = 1)) +
    labs(x="Position", y="Normalized counts") +
    scale_x_continuous(breaks=seq(0, 30000, 5000)) + 
    ggtitle("MHV (+) strand coverage plots")
neg_plot

ggsave(filename="neg_plot.pdf", plot=neg_plot)

#MHV (-) strand coverage plots

data_dir_pos = "~/Dropbox (Hesselberth Lab)/Rachel_data/EndoU_project/viral_bg/pos"
data_files_pos = list.files(data_dir_pos, full.names = T)

pos_table <- purrr::map_df(data_files_pos, read_file) %>%
  mutate(name = str_replace(name, ".mhv.pos.dinuc.bg", "")) %>%
  separate(name, into = c('cell', 'virus', 'time'), sep = '_')

datatable(pos_table)

pos_plot <- ggplot(pos_table, aes(x = start, y = normalized_count)) +
    geom_bar(aes(fill = time), stat = "identity", position = 'dodge') +
    scale_fill_brewer(palette = 'Set1') + 
    theme_cowplot() + 
    facet_grid(virus ~ cell) + 
    theme(axis.text.x = element_text(angle = 60, hjust = 1)) +
    labs(x="Position", y="Counts") +
    scale_x_continuous(breaks=seq(0, 30000, 5000)) +
    ggtitle("MHV (-) strand coverage plots")
pos_plot

```

#Code to generate all the coverage plots for figures

```{r}

neg_table <- neg_table %>%
  filter(time != "012" & time != "09")  %>%
  unite("cell_virus" , c("cell", "virus"), remove = FALSE)

cellvirus_names<-unique(neg_table$cell_virus)

cellvirus_name <- c("B6_MHVS", "B6_MHVV", "B6_mock", "B6_ns2", "B6_nsp15", "IFNAR_MHVS", "IFNAR_MHVV", "IFNAR_mock", "IFNAR_ns2", "IFNAR_nsp15", "RNaseL_MHVS", "RNaseL_MHVV", "RNaseL_mock", "RNaseL_ns2", "RNaseL_nsp15")


for (i in cellvirus_names)
{
plot <- neg_table %>%
  filter(cell_virus == i) %>%  
  mutate(time = fct_relevel(time, "9", "12")) %>%  
  ggplot(aes(x=start, y=normalized_count, fill = time, width = 25)) +
  geom_bar(stat = "identity", position = 'identity') +
  scale_fill_grey(start=0.8, end=0.2) +
  theme_cowplot() + 
  facet_grid(rows = vars(time)) + 
  theme(axis.text.x = element_text(angle = 60, hjust = 1)) +
  labs(x="Position", y="Normalized counts") +
  scale_y_continuous(limits = c(0.00, 0.15)) +
  scale_x_continuous(breaks=seq(0, 30000, 5000)) +
  ggtitle(paste("Coverage plots:",i, sep=""))
  ggsave(paste('~/Dropbox (Hesselberth Lab)/Rachel_data/EndoU_project/images/',i, '.pdf', sep=""), plot)
}


```

#Identifying endonuclease specific cleavage by subtractive analysis

##RNaseL substractive coverage plots of MHV (+) strand reads for all cell types by type of viral infection

The normalized counts detected in RNaseL KO cells for each type of viral infection were subtracted from the normalized counts detected in WT B6 and IFNAR KO cells. This substractive analysis will emphasize RNaseL dependent cleavage events by reporting signals that occur only in the presence of 
RNaseL. 

```{r, message = F, warning = F}

#RNase L substractive analysis to remove signal that occurs in the abscence of RNAse L by cell type 

subtractive_neg <- neg_table %>% 
  dplyr::select(-count, -dinuc) %>% 
  spread(cell, normalized_count) 
subtractive_neg[is.na(subtractive_neg)] <- 0

subtractive_neg <- subtractive_neg %>% 
  mutate(B6_RNaseLdep = B6 - RNaseL) %>% 
  mutate(IFNAR_RNaseLdep = IFNAR - RNaseL) %>% 
  dplyr::select(start:end, virus:time, B6_RNaseLdep:IFNAR_RNaseLdep) %>% 
  gather(cell, normalized_count, B6_RNaseLdep:IFNAR_RNaseLdep)
                                      
#Plots of RNase L dependent sites 
subneg_plot <- ggplot(subtractive_neg, aes(x = start, y = normalized_count)) +
    geom_line(aes(color = time)) + 
    scale_color_brewer(palette = 'Set1') + 
    theme_cowplot() +
    facet_grid(virus ~ cell) + 
    theme(axis.text.x = element_text(angle = 60, hjust = 1)) + 
    labs(x="Position", y="Normalized counts") +
    ylim(0, 0.2) + 
    scale_x_continuous(breaks=seq(0, 30000, 5000)) +
    ggtitle("RNase L dependent-cleavage sites analysis")
subneg_plot

```

#Code to make plots of the subtractive data for the paper with important sites highlighted

```{r}

subtractive_neg_modified <- subtractive_neg %>%
  filter(time != "012" & time != "09" & time != "9" & virus != "mock" )  %>%
  filter(cell == "B6_RNaseLdep") %>%
  unite("cell_virus" , c("cell", "virus"), remove = FALSE)

cellvirus_names<-unique(subtractive_neg_modified$cell_virus)

cellvirus_name <- c("B6_RNaseLdep_MHVS", "B6_RNaseLdep_MHVV", 
                    "B6_RNaseLdep_ns2", "B6_RNaseLdep_nsp15")


for (i in cellvirus_names)
{
plot <-subtractive_neg_modified %>%
  filter(cell_virus == i) %>%  
  #mutate(time = fct_relevel(time, "9", "12")) %>%  
  ggplot(aes(x=start, y=normalized_count, fill = time, width = 25)) +
  geom_bar(stat = "identity", position = 'identity') +
  scale_fill_grey(start=0.2, end=0.8) +
  theme_cowplot() + 
  #facet_grid(rows = vars(time)) + 
  theme(axis.text.x = element_text(angle = 60, hjust = 1)) +
  labs(x="Position", y="Normalized counts") +
  scale_y_continuous(limits = c(0.0, 0.1)) +
  scale_x_continuous(breaks=seq(0, 30000, 5000)) +
  ggtitle(paste("RNaseL-dependent plots:",i, sep=""))
  ggsave(paste('~/Dropbox (Hesselberth Lab)/Rachel_data/EndoU_project/images/',i, '.pdf', sep=""), plot)
}


highlighted_plot <- subtractive_neg %>%
  filter(time != "012" & time != "09" & time != "9" & virus != "mock" )  %>%
  filter(cell == "B6_RNaseLdep") %>%
  filter(virus == "nsp15") %>%
  ggplot(aes(x=start, y=normalized_count, width = 25)) +
  geom_bar(stat = "identity", position = 'identity') + 
  theme_cowplot() + 
  gghighlight(start %in% highlight) +
  theme(axis.text.x = element_text(angle = 60, hjust = 1)) +
  labs(x="Position", y="Normalized counts") +
  scale_y_continuous(limits = c(0.0, 0.2)) +
  scale_x_continuous(breaks=seq(0, 30000, 5000))
highlighted_plot

```


##EndoU mutant (nsp15) substractive coverage analysis of MHV (+) strand for all cell types by type of viral infection

This is very similar to the analysis above, expect it will substract signal occuring in the abscence of nsp15 actviity, which will emphasize sites specific to nsp15 cleavage actviity. 

```{r, message = F, warning = F}

# nsp15 substractive analysis to remove signal that occurs in the abscence of nsp15 activity by viral infection 

subtractive_nsp15 <- neg_table %>% 
  dplyr::select(-count, -dinuc) %>% 
  spread(virus, normalized_count)
subtractive_nsp15[is.na(subtractive_nsp15)] <- 0

subtractive_nsp15 <- subtractive_nsp15 %>% 
  mutate(MHVS_nsp15dep = MHVS - nsp15) %>% 
  mutate(MHVV_nsp15dep = MHVV - nsp15) %>% 
  mutate(ns2_nsp15dep = ns2 - nsp15) %>% 
  dplyr::select(start:end, cell:time, MHVS_nsp15dep:ns2_nsp15dep) %>% 
  gather(virus, normalized_count, MHVS_nsp15dep:ns2_nsp15dep)

#Plots of nsp15 dependent sites 
                         
subnsp15_plot <- subtractive_nsp15 %>%
  filter(cell != "IFNAR") %>%
  filter(cell != "B6") %>% 
  filter(time != "012" & time != "09" & time != "9") %>%
  ggplot(aes(x = start, y = normalized_count, fill = virus, width = 100)) +
  geom_bar(stat = "identity", position = 'identity') +
  scale_fill_brewer(palette="Set1") + 
  theme_cowplot() +
  facet_grid(virus ~ cell) + 
  theme(axis.text.x = element_text(angle = 60, hjust = 1)) + 
  labs(x="Position", y="Normalized counts") +
  ylim(0, 0.2) + 
  scale_x_continuous(breaks=seq(0, 30000, 5000)) +
  ggtitle("Nsp15 dependent-cleavage sites")
subnsp15_plot


ggsave(filename="subnsp15_B6andRNaseL.pdf", plot=subnsp15_plot)


```


#Identifying "significant" sites of nsp15 cleavage

This analysis attempts to calculate which potential cleavage sites have the greatest fold change in cleavage singal captured between samples infected with WT virus and samples infected with nsp15 mutant virus. Cleavage sites where signal captured during WT infection decreases the most during infection with nsp15 mutant virus may be "real" sites of MHV endoU cleavage acitivity. 

Calculating "fold change" across all sites is an unbiased approach to identify potential nsp15 cleavage sites. An alternative approach, focuses on fold change at sites with the greatest signal in the libary. 


##Fold-change for RNaseL 

In this analysis, all positions in the MHV positive strand RNA with more than 20 reads were queried to identify sites with equal to or greater than a 5-fold decrease in signal comparing B6 WT cells and IFNAR KO cells to RNaseL KO cells (B6/RNaseL and IFNAR/RNaseL). 

This analysis is independent of viral infection and just asks which sites change the most when RNaseL is absent. 

```{r}

change_RNaseL <- neg_table %>% 
  filter(count >= 20) %>%
  dplyr::select(-count, -dinuc) %>% 
  spread(cell, normalized_count)
change_RNaseL[is.na(change_RNaseL)] <- 0

change_RNaseL <- change_RNaseL %>% 
  mutate(B6_change = B6/RNaseL) %>% 
  mutate(IFNAR_change = IFNAR/RNaseL) %>% 
  gather(cell_comp, fold_change, B6_change:IFNAR_change) %>%
  filter(fold_change > 0 & fold_change != Inf) %>%
  filter(fold_change >= 5) %>%
  group_by(start) %>%
  unique() %>%
  arrange(desc(fold_change))

datatable(change_RNaseL)

change_RNaseL_graph <- change_RNaseL %>% 
  ggplot(aes(x = start, y = fold_change, fill = cell_comp, width=100)) +
  geom_bar(stat = "identity", position = 'identity') + 
  scale_fill_brewer(palette="Set1") +
  facet_grid(virus ~ cell_comp) +
  theme_cowplot() +
  labs(x="Position", y="fold_change") +
  scale_x_continuous(breaks=seq(0, 30000, 5000)) +
  theme(axis.text.x = element_text(angle = 60, hjust = 1))
change_RNaseL_graph

```


```{r}

change_RNaseL <- neg_table %>% 
  #filter(count >= 20) %>%
  dplyr::select(-count, -dinuc) %>% 
  spread(cell, normalized_count, fill = 0)
change_RNaseL[is.na(change_RNaseL)] <- 0

change_RNaseL <- change_RNaseL %>% 
  mutate(B6_change = (log2(RNaseL/B6))) %>% 
  gather(viral_comp, fold_decrease, B6_change) %>%
  filter(fold_decrease != -Inf & fold_decrease != Inf) %>%
  group_by(start) %>%
  unique() %>%
  arrange((fold_decrease))


common_ns2 <- change_RNaseL %>%
   filter(time == "12") %>%
   filter(virus == "ns2") %>%
   filter(fold_decrease <= -2) %>%
   dplyr::select(start, virus)

common_nsp15 <- change_RNaseL %>%
   filter(time == "12") %>%
   filter(virus == "nsp15") %>%
   filter(fold_decrease <= -1) %>%
   dplyr::select(start, virus)

common_MHVV <- change_RNaseL %>%
   filter(time == "12") %>%
   filter(virus == "MHVV") %>%
   filter(fold_decrease <= -2) %>%
   dplyr::select(start, virus)

common_MHVS <- change_RNaseL %>%
   filter(time == "12") %>%
   filter(virus == "MHVS") %>%
   filter(fold_decrease <= -2) %>%
   dplyr::select(start, virus)


common_final <- left_join(common_ns2, common_nsp15, by = "start")
common_final <- left_join(common_final, common_MHVV, by = "start")
common_final <- left_join(common_final, common_MHVS, by = "start")

highlight = c(26013, 30288, 30290)

highlight <- common_nsp15$start

nsp15_graph <- change_RNaseL %>%
  #mutate(highlight_flag = ifelse(start %in% highlight, T, F)) %>% 
  filter(time == "12") %>%
  filter(virus == "nsp15") %>%
  ggplot(aes(log2(B6), -(fold_decrease))) +
  geom_point() + 
  #geom_point(aes(color = highlight_flag)) +
  gghighlight(start %in% highlight) +
  #gghighlight(fold_decrease < -2.5 , label_key = start) +
  geom_hline(yintercept = 0) + 
  #facet_grid(virus ~ viral_comp) + 
  theme_cowplot()
nsp15_graph



```

##Fold-change for Nsp15 mutant reads

In this analysis, all positions in the MHV positive strand RNA with more than 20 reads were queried to identify sites with equal to or greater than a 5-fold decrease in signal comparing cells infected with WT/ns2 mutant virus to nsp15 mutant virus (MHV WT/nsp15 and ns2/nsp15). 

This analysis is independent of cells type and just asks which sites change the most when nsp15 activity is lost. Thus, these sites are not independent of RNaseL actviity. 

```{r, message = F, warning = F}

change_nsp15 <- neg_table %>% 
  filter(count >= 20) %>%
  dplyr::select(-count, -dinuc) %>% 
  spread(virus, normalized_count)
change_nsp15[is.na(change_nsp15)] <- 0

change_nsp15 <- change_nsp15 %>% 
  mutate(MHVS_change = (log2(nsp15/MHVS))) %>% 
  mutate(MHVV_change = (log2(nsp15/MHVV))) %>% 
  mutate(MHVV_change = (log2(nsp15/mock))) %>% 
  mutate(ns2_change = (log2(nsp15/ns2))) %>% 
  gather(viral_comp, fold_decrease, MHVS_change:ns2_change) %>%
  filter(fold_decrease != -Inf & fold_decrease != Inf) %>%
  #filter(fold_decrease >= 80) %>%
  group_by(start) %>%
  unique() %>%
  arrange((fold_decrease))

datatable(change_nsp15)

change_nsp15_graph <- change_nsp15 %>%
  ggplot(aes(x = start, y = fold_change, fill = viral_comp, width=100)) +
  geom_bar(stat = "identity", position = 'identity') + 
  scale_fill_brewer(palette="Set1") +
  facet_grid(cell ~ viral_comp) +
  theme_cowplot() +
  labs(x="Position", y="fold_change") +
  scale_x_continuous(breaks=seq(0, 30000, 5000)) +
  theme(axis.text.x = element_text(angle = 60, hjust = 1))
change_nsp15_graph


nsp15_graph <- change_nsp15 %>%
  filter(time == "12") %>%
  filter(cell != "IFNAR") %>%
  filter(cell != "RNaseL") %>%
  ggplot(aes(MHVV, fold_decrease)) +
  geom_point() + 
  gghighlight(fold_decrease < -3.5 , label_key = start) +
  geom_hline(yintercept = 0) + 
  theme_cowplot()
nsp15_graph
                      
ggsave(filename="MHVV_B6_foldchange.pdf", plot=nsp15_graph)



```

##Fold-change for Nsp15 muant and RNaseL KO reads

Same analysis as above focused in RNaseL KO cells. 

```{r}

change_nsp15 <- neg_table %>% 
  filter(count >= 20) %>%
  dplyr::select(-count, -dinuc) %>% 
  spread(virus, normalized_count)
change_nsp15[is.na(change_nsp15)] <- 0

change_nsp15RNaseL <- change_nsp15 %>% 
  mutate(MHVS_change = MHVS/nsp15) %>% 
  mutate(MHVV_change = MHVV/nsp15) %>% 
  mutate(ns2_change = ns2/nsp15) %>% 
  gather(viral_comp, fold_change, MHVS_change:ns2_change) %>%
  filter(fold_change > 0 & fold_change != Inf) %>%
  filter(fold_change >= 5) %>%
  filter(cell == "RNaseL") %>%
  group_by(start) %>%
  unique() %>%
  arrange(desc(fold_change))

datatable(change_nsp15RNaseL)

change_nsp15RNaseL_graph <- change_nsp15RNaseL %>%
  ggplot(aes(x = start, y = fold_change, fill = viral_comp, width=100)) +
  geom_bar(stat = "identity", position = 'identity') + 
  scale_fill_brewer(palette="Set1") +
  facet_grid(cell ~ viral_comp) +
  theme_cowplot() +
  labs(x="Position", y="fold_change") +
  scale_x_continuous(breaks=seq(0, 30000, 5000)) +
  theme(axis.text.x = element_text(angle = 60, hjust = 1))
change_nsp15RNaseL_graph


```

##Top RNaseL-dependent sites

In this analysis, the sites with the greatest normalized counts that are RNaseL dependent are identified. This relies on an inital subtractive analysis to emphasize signal that  occurs when RNaseL is present. The top 50 sites by cell type that are RNaseL dependent are then identified. This data is filtered by sites with 20 or more reads. 

```{r, message = F, warning = F}

top_RNaseL <- neg_table %>% 
  filter(count >= 20) %>%
  dplyr::select(-count, -dinuc) %>% 
  spread(cell, normalized_count) 
top_RNaseL[is.na(top_RNaseL)] <- 0

top_RNaseL <- top_RNaseL %>% 
  mutate(B6_RNaseLdep = B6 - RNaseL) %>% 
  mutate(IFNAR_RNaseLdep = IFNAR - RNaseL) %>% 
  dplyr::select(start:end, virus:time, B6_RNaseLdep:IFNAR_RNaseLdep) %>% 
  gather(cell, normalized_count, B6_RNaseLdep:IFNAR_RNaseLdep) %>%
  group_by(cell) %>%
  arrange(desc(normalized_count)) %>%
  do(head(., n = 50))
  
datatable(top_RNaseL)

top_RNaseL_graph <- 
  ggplot(top_RNaseL, aes(x = start, y = normalized_count, fill = cell, width=100)) +
  geom_bar(stat = "identity", position = 'identity') + 
  scale_fill_brewer(palette="Set1") +
  facet_grid(virus ~ cell) +
  theme_cowplot() +
  labs(x="Position", y="normalized_count") +
  scale_x_continuous(breaks=seq(0, 30000, 5000)) +
  theme(axis.text.x = element_text(angle = 60, hjust = 1))
top_RNaseL_graph

```


##Top nsp15-dependent sites

In this analysis, the sites with the greatest normalized counts that are nsp15-dependent are identified. This relies on an inital subtractive analysis to emphasize signal that  occurs when nsp15 is present. The top 50 sites by cell type that are nsp15-dependent are then identified. 


```{r}

top_nsp15 <- neg_table %>% 
  filter(count >= 20) %>%
  dplyr::select(-count, -dinuc) %>% 
  spread(virus, normalized_count)
top_nsp15[is.na(top_nsp15)] <- 0

top_nsp15 <- top_nsp15 %>% 
  mutate(MHVS_nsp15dep = MHVS - nsp15) %>% 
  mutate(MHVV_nsp15dep = MHVV - nsp15) %>% 
  mutate(ns2_nsp15dep = ns2 - nsp15) %>% 
  dplyr::select(start:end, cell:time, MHVS_nsp15dep:ns2_nsp15dep) %>% 
  gather(virus, normalized_count, MHVS_nsp15dep:ns2_nsp15dep) %>%
  group_by(virus) %>%
  arrange(desc(normalized_count)) %>%
  do(head(., n = 50))

datatable(top_nsp15)

top_nsp15_graph <- 
  ggplot(top_nsp15, aes(x = start, y = normalized_count, fill = virus, width = 100)) +
  geom_bar(stat = "identity", position = 'identity') + 
  scale_fill_brewer(palette="Set1") +
  facet_grid(cell ~ virus) +
  theme_cowplot() +
  labs(x="Position", y="normalized_count") +
  scale_x_continuous(breaks=seq(0, 30000, 5000)) +
  theme(axis.text.x = element_text(angle = 60, hjust = 1))
top_nsp15_graph

```


#Calculating dinucelotide frequencies for all cell types by infection status 

Performing a dinucleotide frequency analysis will help us to confirm RNaseL signatures as a positive control and investigate a potential sequence preference for nsp15 cleavage.

```{r, message = F, warning = F}

# Calculating freuqency of each dinucleotide for the + strand MHV reads 

dinuc <- c("AA","GA", "CA", "UA","AC", "CC", "GC", "UC", "AG", "CG", "GG", "UG", "AU", "CU", "GU", "UU")
df_dinuc <- data_frame(dinuc)

frequencies_mhv <- readr::read_tsv("mhv_fasta_dinuc_counts.txt") %>%
  dplyr::mutate(frequency = count/sum(count)) %>%
  rename(mhv_count = count) %>%
  rename(mhv_freq = frequency)

frequencies_neg <- function(x) {
  df <- readr::read_tsv(x, col_names = c("chrom", "start", "end", "count",      "normalized_count", "dinuc"))
  df <- aggregate(cbind(count) ~ dinuc, data = df, sum)
  df <- left_join(df_dinuc, df) %>% 
    replace_na(list(count = 0)) %>% 
    #left_join(frequencies_mhv, by = "dinuc") %>%
    mutate(frequency = count/sum(count))
  df$name <- basename(x)
  df %>% mutate(name = str_replace(name, ".mhv.neg.dinuc.bg", "")) %>%
  separate(name, into = c('cell', 'virus', 'time'))
}


neg_freq <- purrr::map_df(data_files_neg, frequencies_neg) %>%
  left_join(frequencies_mhv) %>%
  mutate(final_frequency = frequency/mhv_freq)


#mutate(frequency = count/(sum(mhv_count)+sum(count)))

```

##Plotting dinucleotide frequencies for all cell types by type of viral infection

These plots show the freuqency of cleavege at 3´-dinucleotides for the MHV (+) sense RNA from highest to lowest frequency. I removed the mock infected sample data from these graphs. 

```{r, message = F, warning = F}

# Dincleotide freuqnecy plots 

neg_freq_plot <- neg_freq %>%
  filter(time == "12") %>%
  filter(cell != "IFNAR") %>%
  filter(virus != "MHVS") %>%
  mutate(virus = fct_relevel(virus, "mock", "MHVV", "ns2", "nsp15")) %>%
  ggplot(aes(x = dinuc , y = final_frequency)) + 
  scale_fill_brewer(palette = 'Set1') + 
  scale_x_discrete(limits=c("AA","GA", "CA", "UA","AC", "CC", "GC", "UC", "AG", "CG", "GG",   "UG", "AU", "CU", "GU", "UU")) + 
  geom_bar(aes(fill = virus), stat = "identity", position = 'dodge') + 
  theme_cowplot() + 
  facet_grid(virus ~ cell) + 
  #scale_y_continuous(limits = c(0.00, 0.3)) +
  theme(axis.text.x = element_text(angle = 60, hjust = 1, size=8)) +
  ggtitle("MHV (+) strand dinucleotide frequency plots")
neg_freq_plot


ggsave(filename="MHV_dinuc.pdf", plot=neg_freq_plot)

```

## RNase L substractive dinucleotide analysis of for all cell types by type of viral infection

This is similar to the previous coverage substractive analysis. These plots display dinucleotde cleavage frequencies dependent on RNase L by removing the frequencies detected in the abscence of RNase L. 

```{r, message = F, warning = F}

# Subtractive RNaseL dinucleotide analysis

subtractive_neg <- neg_freq %>% 
  dplyr::select(dinuc, frequency:time) %>% 
  spread(cell, frequency) %>% 
  mutate(B6_RNaseLdep = B6 - RNaseL) %>% 
  mutate(IFNAR_RNaseLdep = IFNAR - RNaseL) %>% 
  dplyr::select(dinuc:time, B6_RNaseLdep:IFNAR_RNaseLdep) %>% 
  gather(cell, frequency, B6_RNaseLdep:IFNAR_RNaseLdep)

# Plots 

subneg_freq_plot <- subtractive_neg %>% 
  filter(time != "012") %>% 
  filter(time != "09") %>% 
  filter(time != "9") %>%
  filter(cell == "B6_RNaseLdep") %>%
  filter(virus != "MHVS" & virus != "ns2") %>%
  mutate(virus = fct_relevel(virus, "mock", "MHVV", "ns2")) %>%
  ggplot(aes(x = dinuc , y = frequency)) + 
  scale_fill_brewer(palette = 'Set1') + 
  scale_x_discrete(limits=c("AA","GA", "CA", "UA","AC", "CC", "GC", "UC", "AG", "CG", "GG",     "UG", "AU", "CU", "GU", "UU")) + 
  geom_bar(aes(fill = virus), stat = "identity", position = 'dodge') + 
  theme_cowplot() +
  facet_grid(virus ~ cell) + 
  scale_y_continuous(limits = c(0.00, 0.3)) +
  theme(axis.text.x = element_text(angle = 60, hjust = 1, size = 8)) +
  ggtitle("MHV (+) strand RNase L subtractive dinucleotide frequency analysis")
subneg_freq_plot

ggsave(filename="subneg_freq_plot.pdf", plot=subneg_freq_plot)


```

## Nsp15 substractive dinucleotide analysis of for all cell types by type of viral infection

These plots indentify some dinucleotide cleavage events that might be dependent on nsp15 activity and independent of RNase L activity. 

```{r, message = F, warning = F}

#Subtractive nsp15 dinucleotide analysis

subdinuc_neg_nsp15 <- neg_freq %>% 
  dplyr::select(dinuc, frequency:time) %>% 
  spread(virus, frequency) %>%
  mutate(MHVS_nsp15dep = MHVS - nsp15) %>% 
  mutate(MHVV_nsp15dep = MHVV - nsp15) %>% 
  mutate(mock_nsp15dep = mock - nsp15) %>% 
  mutate(ns2_nsp15dep = ns2 - nsp15) %>% 
  dplyr::select(dinuc:time, MHVS_nsp15dep:ns2_nsp15dep) %>% 
  gather(virus, frequency, MHVS_nsp15dep:ns2_nsp15dep)

# Plots

subnegdinuc_freq_plot <- subdinuc_neg_nsp15 %>% 
  filter(time != "012") %>% 
  filter(time != "09") %>% 
  filter(time != "9") %>%
  filter(cell == "RNaseL") %>%
  filter(virus != "MHVS_nsp15dep" & virus != "ns2_nsp15dep") %>%
  mutate(virus = fct_relevel(virus, "mock_nsp15dep", "MHVV_nsp15dep")) %>%
  ggplot(aes(x = dinuc , y = frequency)) + 
  scale_fill_brewer(palette = 'Set1') + 
  scale_x_discrete(limits=c("AA","GA", "CA", "UA","AC", "CC", "GC", "UC", "AG", "CG", "GG",   "UG", "AU", "CU", "GU", "UU")) + 
  geom_bar(aes(fill = virus), stat = "identity", position = 'dodge') + 
  theme_cowplot() +
  facet_grid(virus ~ cell) + 
  theme(axis.text.x = element_text(angle = 60, hjust = 1, size = 8)) +
  scale_y_continuous(limits = c(0.00, 0.10)) +
  ggtitle("MHV (+) strand nsp15 dependent dinucleotide frequencies")
subnegdinuc_freq_plot

ggsave(filename="nsp15_dep_dinuc.pdf", plot=subnegdinuc_freq_plot)

```

